### ISSUE TITLE
Confusing function loader type names

### ISSUE THREAD

**Issue by ethan-tqa** [MaikKlein/ash#15]
So we have a Swapchain type that acts as a function loader, then there is an actual Vulkan type called SwapchainKHR. I think this creates unnecessary confusion, and the type can be named SwapchainLoader or something instead. This applies to other loader types as well. Example of the problem:
`let swapchain_loader = Swapchain::new(&instance, &device).unwrap();`
`let swapchain = swapchain_loader.create_swapchain_khr(&swapchain_create_info, None).unwrap();`

By the way, is it okay to create this loader object multiple times during the application lifetime?

**Comment by ethan-tqa** [comment]
While we are at this subject, I think it might be better to combine function loader and vulkan handle into one type, and it would automatically convert to raw handle, similar to official C++ wrapper that comes with the Vulkan SDK. That way the API will be much more ergonomic.

**Comment by MaikKlein** [comment]
> By the way, is it okay to create this loader object multiple times during the application lifetime?

Yes you can create it as often as you want as long as the device is valid. It also implements clone. 

>  ... the type can be named SwapchainLoader or something instead.

I see your point, I guess that can be confusing.

**Comment by MaikKlein** [comment]
>While we are at this subject, I think it might be better to combine function loader and vulkan handle into one type, and it would automatically convert to raw handle, similar to official C++ wrapper that comes with the Vulkan SDK. That way the API will be much more ergonomic.

You mean something like this?

```
struct Swapchain{
    loader: SwapchainLoader,
    handle: vk::SwapchainKHR
}
```

This would imply that you would always have to clone both. I wanted ash to be as similar to raw Vulkan as possible. 

The official vulkan sdk doesn't have such fine grained control, it just exposes everything in the device itself. (Afaik, Although that might have changed).

I don't see any big ergonomic improvement of combining them.

**Comment by ethan-tqa** [comment]
Yeah, sort of like that. The wrapper enhancements is mainly for the instance, physical device and device types. So for example I can do these in a very handy manner (C++):
`instance = createInstance(ici);`
`surface = instance.createWin32SurfaceKHR(sci);`
`device = physicalDevice.createDevice(dci);`
`queueFamProps = physicalDevice.getQueueFamilyProperties();`
